### PR TITLE
 PIA-486: As a loyalty partner, I would like to track events on iOS.

### PR DESCRIPTION
**Pull request by maciejtrybilo**
Updated all of the `guard let `self` = self` in a separate commit.

### CODE REVIEW

File: GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift [hunk 1]
```diff
@@ -196,6 +199,16 @@ extension GiniScreenAPICoordinator {
 extension GiniScreenAPICoordinator {
     
     @objc func back() {
+        
+        switch screenAPINavigationController.topViewController {
+        case is CameraViewController:
+            trackingDelegate?.onCameraScreenEvent(event: Event(type: .exit))
```
Comment (a-szotyori) [inline GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift]: Documentation should mention, that `.exit` is triggered only when using the Screen API.

File: GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift [hunk 2]
```diff
@@ -196,6 +199,16 @@ extension GiniScreenAPICoordinator {
...
+        case is AnalysisViewController:
+            trackingDelegate?.onAnalysisScreenEvent(event: Event(type: .cancel))
```
Comment (a-szotyori) [inline GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift]: Documentation should mention, that `.cancel` is triggered only when using the Screen API.

File: GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift [hunk 3]
```diff
@@ -208,6 +221,9 @@ extension GiniScreenAPICoordinator {
     }
     
     @objc func showHelpMenuScreen() {
+        
+        trackingDelegate?.onCameraScreenEvent(event: Event(type: .help))
```
Comment (a-szotyori) [inline GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift]: Documentation should mention, that `.help` is triggered only when using the Screen API.

File: GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift [hunk 4]
```diff
@@ -225,6 +241,12 @@ extension GiniScreenAPICoordinator {
     }
     
     @objc func showAnalysisScreen() {
+        
+        if screenAPINavigationController.topViewController is MultipageReviewViewController ||
+        screenAPINavigationController.topViewController is ReviewViewController {
+            trackingDelegate?.onReviewScreenEvent(event: Event(type: .next))
```
Comment (a-szotyori) [inline GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift]: `.next` could go into the `ReviewViewController` and then it's also triggered when using the Component API.

File: GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator+Review.swift [hunk 5]
```diff
@@ -90,6 +90,9 @@ extension GiniScreenAPICoordinator: MultipageReviewViewControllerDelegate {
     }
     
     @objc fileprivate func closeMultipageScreen() {
+        
+        trackingDelegate?.onReviewScreenEvent(event: Event(type: .back))
```
Comment (a-szotyori) [inline GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator+Review.swift]: Documentation should mention, that `.back` is triggered only when using the Screen API.

File: Documentation/source/Event tracking guide.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+The version 5.2 of Gini Vision Library introduces the ability to track user events in the Screen API. In order to receive the events, implement the `GiniVisionTrackingDelegate` protocol and supply the delegate when initializing GVL. For example:
+
+```swift
+let viewController = GiniVision.viewController(withClient: client,
+                                               importedDocuments: visionDocuments,
+                                               configuration: visionConfiguration,
+                                               resultsDelegate: self,
+                                               documentMetadata: documentMetadata,
+                                               trackingDelegate: trackingDelegate)
```
Comment (a-szotyori) [inline Documentation/source/Event tracking guide.md]: You should also show how to add the tracking delegates when using the Component API.

File: GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift (hunk 4 above)
Comment (maciejtrybilo) [reply GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift]: I'm not sure you can. The Next button is managed by the coordinator.

Comment (a-szotyori) [reply GiniVision/Classes/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift]: I see, of course, the next button is in the navigation bar, forgot about that.

**Comment by maciejtrybilo** [comment]
Thanks, @a-szotyori. Have a look!

**Pull request merged**